api/infrastructure/parallelcluster-api.yaml

```diff
@@ -221,6 +221,103 @@ Resources:
       ImageTestsConfiguration:
         ImageTestsEnabled: false
 
+  EcrImageDeletionLambda:
+    Condition: DoNotUseCustomEcrImageUri
+    Type: AWS::Lambda::Function
+    Properties:
+      Code:
+        ZipFile: |
+          import cfnresponse
+          import boto3
+
+          def get_tag_from_image_arn(image_arn):
+              """ Returns the Docker image tag given an ImageBuilder image ARN
+
+              Arn format: arn:aws:imagebuilder:${AWS::Region}:${AWS::AccountId}:image/.../${Version}/${BuildNumber}
+              Tag format: ${Version}-${BuildNumber}
+              """
+              parts = image_arn.split('/')
+              version = parts[-2]
+              build_number = parts[-1]
+              return '{}-{}'.format(version, build_number)
+
+          def handle_deletion_event(event):
+              """ Handle deletion event by removing the created Docker image from the ECR repository
+              """
+              ecr = boto3.client('ecr')
+              ecr_repository_name = event['ResourceProperties']['EcrRepositoryName']
+              tag = get_tag_from_image_arn(event['ResourceProperties']['ImageToDeleteArn'])
+              return ecr.batch_delete_image(repositoryName=ecr_repository_name, imageIds=[{'imageTag': tag}])
+
+          def handler(event, context):
+              print(event)
+              print('boto version {}'.format(boto3.__version__))
+
+              response_data = {}
+              response_status = cfnresponse.SUCCESS
+
+              if event['RequestType'] == 'Create':
+                  response_data['Message'] = 'Resource creation successful!'
+              elif event['RequestType'] == 'Update':
+                  response_data['Message'] = 'Resource update successful!'
+              elif event['RequestType'] == 'Delete':
+                  try:
+                      handle_deletion_event(event)
+                      response_data['Message'] = 'Resource deletion successful!'
+                  except Exception as exception:
+                      response_status = cfnresponse.FAILURE
+                      response_data['Message'] = exception
```

Comment:
Can we prepend something to the exception message such as "Failed when deleting bla bla with error bla"?